### PR TITLE
Use GL_CLAMP_TO_BORDER instead of GL_CLAMP_TO_EDGE

### PR DESCRIPTION
**Pull request by WaywardHeart**
What the title says, although our version of OpenGL is technically calling GL_CLAMP_TO_BORDER_NV. I think this only makes a difference if a game calls stretch_blt with a missized sourceRect. Without this fix, the outermost pixels of the source bitmap will get stretched to the edge of the blitted area. With it, the empty area will be transparent and the destination will keep what was originally there, which matches RPG Maker's behavior.

### CODE REVIEW

**Comment by Splendide-Imaginarius** [comment]
Very nice detective work! This bug had been annoying me for ages and I never figured out what was causing it. Give me a few days to verify the fix, then I'll merge if everything looks OK.

**Comment by Splendide-Imaginarius** [comment]
I can confirm that this fixes the `stretch_blt` bug for me.

@Ancurio does this fix look OK to you? You know OpenGL a lot better than I do, so maybe you can better gauge whether this could cause unwanted side effects.

**Comment by Ancurio** [comment]
From grepping the headers, it appears `GL_CLAMP_TO_BORDER` is legal in desktop GL since 1.3, so it's not an issue using it here. As for GLES2, it appears to only be supported in case one of three extensions are present, `GL_CLAMP_TO_BORDER_{OES,EXT,NV}` (note: I did not check spec differences between the 3, I assume they all do roughly the same).

Since mkxp currently advertises full support for running on GLES2, that support would have to be retracted. Personally, I feel like using a texture wrapping mode for something that can be fixed purely in software with a bit of math to be a workaround, even though it saves on code, but this isn't my fork, so :shrug: 

As for the commit itself, I think it is not correct, as the bug is *still present* in cases where the used GL header simply lacks the define, which is not even what the commit is intended to do? (I would assume the intention was to only use border clamping in cases the driver supports it, which is not done via compile-time define checking but via runtime extension checking; [this set](https://github.com/mkxp-z/mkxp-z/blob/dev/src/display/gl/gl-fun.cpp#L123) can be queried for this. If compilation with gles2 headers is desired to still be supported, just put `#define GL_CLAMP_TO_BORDER 0x812D` [here](https://github.com/mkxp-z/mkxp-z/blob/dev/src/display/gl/gl-fun.h#L117) instead of all the #ifdefing, and definitely decide at runtime which constant to actually use based on available extensions.

So, in short, this PR, aside the code issues, fixes the issue on all desktop systems and certain mobile systems. A purely mkxp-side fix involving a bit of math would fix it everywhere, but introduce more code.

**Comment by WaywardHeart** [comment]
Back when I wrote that, I decided to include fallbacks so that things would keep working without a fuss if the GL version we're using changed, and chose to use GL_CLAMP_TO_BORDER as the name of the macro because I thought of it as the "proper" name for it. Thinking about it further, that was kind of silly of me. If a change like that happens, we'll want to actually know about it, and doing that with the macro names just added confusion. I just pushed a commit to use the proper GL_CLAMP_TO_BORDER_NV macro.

**Comment by Splendide-Imaginarius** [comment]
> Personally, I feel like using a texture wrapping mode for something that can be fixed purely in software with a bit of math to be a workaround

The texture wrapping mode would yield better performance than a software fix, right?

**Comment by Ancurio** [comment]
> > Personally, I feel like using a texture wrapping mode for something that can be fixed purely in software with a bit of math to be a workaround
> 
> The texture wrapping mode would yield better performance than a software fix, right?

Not really. The software fix is to properly clamp the source rect if it extends beyond the bitmap dimensions, and translate this clamping to the dest rect so proportions are conserved.

File: src/display/gl/gl-util.h [hunk 1]
```diff
@@ -93,8 +93,8 @@ namespace TEX
 
 	static inline void setRepeat(bool mode)
 	{
-		gl.TexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, mode ? GL_REPEAT : GL_CLAMP_TO_EDGE);
-		gl.TexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, mode ? GL_REPEAT : GL_CLAMP_TO_EDGE);
+		gl.TexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, mode ? GL_REPEAT : GL_CLAMP_TO_BORDER_NV);
+		gl.TexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, mode ? GL_REPEAT : GL_CLAMP_TO_BORDER_NV);
```
Comment (Ancurio) [inline src/display/gl/gl-util.h:97]: Doesn't this still break compilation with GLES2 headers?

**Comment by Splendide-Imaginarius** [comment]
> > > Personally, I feel like using a texture wrapping mode for something that can be fixed purely in software with a bit of math to be a workaround
> > 
> > 
> > The texture wrapping mode would yield better performance than a software fix, right?
> 
> Not really. The software fix is to properly clamp the source rect if it extends beyond the bitmap dimensions, and translate this clamping to the dest rect so proportions are conserved.

@WaywardHeart Would you be willing to try implementing the software fix approach that @Ancurio described?

**Comment by WaywardHeart** [comment]
Yeah, doing it that way sounds better now. Closing this PR.
